components/channel_members_rhs/member.tsx

```diff
@@ -99,6 +100,14 @@ interface Props {
     };
 }
 
+function isChannelAdmin(channelMember: ChannelMembership | undefined) {
+    if (!channelMember) {
+        return false;
+    }
+
+    return isChannelAdminUtils(channelMember.roles) || channelMember.scheme_admin;
```

Comment:
When does it come up that the channel member has `scheme_admin` set to true, but they're not an admin according to their roles? Is that something that's broken everywhere that we're only checking the user's roles?

Reply:
I reused this from the previous management modal so I'm not 100% why it is like this, but looking at the server code I could find that when using custom role schemes, `roles` could not be set? `scheme_admin` seems to always be present, maybe I could rely on this one only 🤔

Reply:
source: https://github.com/mattermost/mattermost-server/blob/master/app/channel.go#L1145

Reply:
That seems really odd to me, but I don't really understand the in-and-outs of the permissions system. We definitely have other places where we only look at the roles to determine if someone's an admin like [in the Telemetry code](https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/client/rudder.ts#L62) and a lot of other places we use those `isXAdmin` helpers from mattermost-redux.

I don't think anything needs to change here though, but this does have me worried that there's cases we're not correctly identifying admins elsewhere